### PR TITLE
added user route to get users profile by his unique id

### LINKED ISSUE #20: login/sign-up and logout APIs
We are introducing a login/sign-up functionality for users using passport and jwt-authentication. This feature will introduce many other possibilities we can provide to respective users. Follow the REST architectural style for implementing these APIs.

### PR DESCRIPTION
**Pull request by ChandanaReddy07**
Fixes #20
Added user rout having property to get users profile by his unique id, added isSignedin and isAuthenticated featurs in auth controller, added validation middleweres by using express validator in signin and signout routes

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
 const express = require("express");
+const app = express();
```
Comment (arjungarg07) [inline index.js]: Don't create the app instance directly, Here firstly all the required libraries will be mentioned

**Review by arjungarg07 — CHANGES_REQUESTED** [review]

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "configurations": [
+        {
+            "type": "pwa-chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (arjungarg07) [inline .vscode/launch.json]: is this file necessary? Remove it if it works without this.

Comment (arjungarg07) [reply .vscode/launch.json]: @dcr7978933k have a look at it.

File: controllers/auth.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+//custom middlewares
+exports.isAuthenticated = (req,res,next)=>{
+  let checker =req.profile&&req.auth&&req.profile._id==req.auth._id
+  if(!checker){
+    res.status(403).json({
+      error:"ACCESS DINIED"
+    })
+  }
+  next();
+};
```
Comment (arjungarg07) [inline controllers/auth.js]: Can we simply make the functions async, use try,catch block and export all of them at the bottom of the file?
It would be convenient to have it that way.

File: models/User.js [hunk 4]
```diff
@@ -1,15 +1,63 @@
 const mongoose = require('mongoose');
-const { Schema } = mongoose;
+const crypto = require('crypto');
+const uuidv1 = require('uuid/v1');
```
Comment (arjungarg07) [inline models/User.js:3]: @dcr7978933k  What's the use case for this?

**Review by arjungarg07 — CHANGES_REQUESTED** [review]

File: models/User.js (hunk 4 above)
Comment (ChandanaReddy07) [reply models/User.js:3]: Crypto is used here for encrypting the password and we can use it directly with Node.js so no need of npm install
and uuid is a simple and a faster way of creating a longer string ,generating unique id for hashing the password is one of the use case of it.

File: controllers/auth.js (hunk 3 above)
Comment (ChandanaReddy07) [reply controllers/auth.js]: yes made the changes.

File: .vscode/launch.json (hunk 2 above)
Comment (ChandanaReddy07) [reply .vscode/launch.json]: There was no need of that file . So deleted
